staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -707,12 +707,12 @@ func (qs *queueSet) findDispatchQueueLocked() *queue {
 		queue := qs.queues[qs.robinIndex]
 		oldestWaiting, _ := queue.requests.Peek()
 		if oldestWaiting != nil {
-			sMin = math.Min(sMin, queue.virtualStart)
-			sMax = math.Max(sMax, queue.virtualStart)
-			estimatedWorkInProgress := qs.estimatedServiceSeconds * float64(queue.seatsInUse)
-			dsMin = math.Min(dsMin, queue.virtualStart-estimatedWorkInProgress)
-			dsMax = math.Max(dsMax, queue.virtualStart-estimatedWorkInProgress)
-			currentVirtualFinish := queue.virtualStart + qs.estimatedServiceSeconds*float64(oldestWaiting.Seats())
+			sMin = math.Min(sMin, queue.nextDispatchR.ToFloat())
```

Comment:
Please don't convert to floats - the whole point I want this to be int is to not loose prevision on math operations.

Reply:
`sMin` is here only to compute a Prometheus gauge value (which is necessarily a float64, right?).

Reply:
I know - but still. I would like all this code to operate on int.

[And the only potential conversions to floats happening in the metrics code]

Reply:
Deferred the conversion to floating-point to the metrics call.